cypress/plugins/nodeManager/plugin.ts

```diff
@@ -108,25 +111,34 @@ class Node {
     this.httpPort = options.id;
     this.peerPort = options.id;
     this.proxyBinaryPath = path.join(ROOT_PATH, options.proxyBinaryPath);
+    this.storagePath = path.join(CYPRESS_WORKSPACE_PATH, uuid.v4());
   }
 
   async start() {
     this.logger.log("starting node");
 
+    fs.mkdirsSync(this.storagePath);
```

Comment:
We should avoid using blocking code
```suggestion
    await fs.promises.mkdir(this.storagePath);
```

Reply:
See: https://github.com/radicle-dev/radicle-upstream/pull/1589/files#r574708931.

Reply:
I don’t think your comment applies since we are not calling `start()` from the sync exit hook. The `start` method is already async.

Reply:
I still think it does.

In the `start` method we register a callback that calls `cleanup` when the process exits: https://github.com/radicle-dev/radicle-upstream/blob/rudolfs/replication-tests/cypress/plugins/nodeManager/plugin.ts#L361

And since we run the `stop` method for each node in the `exitHook` that code will be indirectly invoked.

Reply:
Hm, you're right. The process exit callback never fires : |

Reply:
This seems to fix our problem: 9eec21f550be4b4a033be67f6ecf74650b8accf0.